projects/sbb-esta/angular-business/chip/src/chip-input/chip-input.component.ts

```diff
@@ -0,0 +1,260 @@
+   * @docs-private
+   */
+  @Output() readonly valueChange: EventEmitter<any> = new EventEmitter<any>();
+
+  focused = false;
+  errorState: boolean;
+  inputModel = '';
+  origin = new AutocompleteOriginDirective(this._elementRef);
+  selectionModel: SelectionModel<string>;
+
+  private _onTouchedCallback: () => void;
+  private _onChangeCallback: (_: any) => void;
+
+  /** Whether filling out the chip-input is required in the form. */
+  private _required = false;
+
+  /** The aria-describedby attribute on the chip-input for improved a11y. */
+  private _ariaDescribedby: string;
+
+  /** Unique id for this input. */
+  private _uid = `sbb-chip-input-${nextUniqueId++}`;
+
+  /** Emits when the state of the option changes and any parents have to be notified. */
+  readonly stateChanges = new Subject<void>();
+
+  constructor(
+    @Self() @Optional() public ngControl: NgControl,
+    private _elementRef: ElementRef,
+    private _changeDetectorRef: ChangeDetectorRef
+  ) {
+    if (this.ngControl) {
+      // Note: we provide the value accessor through here, instead of
+      // the `providers` to avoid running into a circular import.
+      this.ngControl.valueAccessor = this;
+    }
+
+    // Force setter to be called in case id was not specified.
+    this.id = this.id;
+  }
+
+  ngOnInit(): void {
+    this.selectionModel = new SelectionModel<string>(true);
+    this._initializeSelection();
+    if (this.autocomplete) {
+      this.autocomplete.optionSelected.subscribe(event => this.onSelect(event.option.value));
+    }
+  }
+
+  /**
+   * Sets the chip input's value. Part of the ControlValueAccessor interface
+   * required to integrate with Angular's core forms API.
+   *
+   * @param value New value to be written to the model.
+   */
+  writeValue(value: string[]): void {
+    if (this.selectionModel) {
+      value.forEach(v => this.onSelect(v));
+    }
+  }
+
+  /**
+   * Disables the chip input. Part of the ControlValueAccessor interface required
+   * to integrate with Angular's core forms API.
+   *
+   * @param isDisabled Sets whether the component is disabled.
+   */
+  setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+    this._changeDetectorRef.markForCheck();
+    this.stateChanges.next();
+  }
+
+  /**
+   * Adds a given value to the current selected values.
+   */
+  onSelect(option: string) {
+    if (!this.selectionModel.isSelected(option)) {
+      this.selectionModel.select(option);
+      this._propagateChanges();
+    }
+    this.inputElement.nativeElement.value = '';
+  }
+
+  /**
+   * Removes a given value from the current selected values.
+   */
+  deselectOption(option: string) {
+    if (this.selectionModel.isSelected(option)) {
+      this.selectionModel.deselect(option);
+      this._propagateChanges(option);
+    }
+  }
+
+  /** `View -> model callback called when value changes` */
+  onChange: (value: any) => void = () => {};
```

Comment:
I think this can be removed? This is handle by _onChangeCallback?

Reply:
Yep it can 🙃